### PR TITLE
#Add cors config for dev & prod web domain

### PR DESCRIPTION
**Pull request by nets7snipper**
Test result- https://q240iu43yr.feishu.cn/docx/RUUSdp6bNow5Ynxmv5XckCWNnVf

### CODE REVIEW

File: blotztask-api/Program.cs [hunk 1]
```diff
@@ -56,12 +56,16 @@
...
         {
-            builder.AllowAnyOrigin()
-                   .AllowAnyMethod()
-                   .AllowAnyHeader();
+            builder.WithOrigins("http://localhost:3000" // DEV frontend origin
+                , "https://blotz-task-app.vercel.app") // Prod frontend origin    
+                .WithMethods("GET", "POST", "OPTIONS") // Specify allowed methods, do not allow method never used.
+                .WithHeaders("Content-Type", "Authorization") // Specify allowed headers,may be more headers to added.
+                .WithExposedHeaders("X-Custom-Header") // Specify biz headers, x-custom-header as an example.
+                .AllowCredentials(); // anti-csrf need to be built.
```
Comment (sol-wizard) [inline blotztask-api/Program.cs:68]: do you mean TODO: anti-csrf need to be built.? you can add a TODO keyword infront ,

**Review by sol-wizard — DISMISSED** [review]

File: blotztask-api/Program.cs [hunk 2]
```diff
@@ -56,12 +56,16 @@
...
         builder =>
         {
-            builder.AllowAnyOrigin()
-                   .AllowAnyMethod()
-                   .AllowAnyHeader();
+            builder.WithOrigins("http://localhost:3000" // DEV frontend origin
+                , "https://blotz-task-app.vercel.app") // Prod frontend origin    
+                .WithMethods("GET", "POST", "OPTIONS") // Specify allowed methods, do not allow method never used.
+                .WithHeaders("Content-Type", "Authorization") // Specify allowed headers,may be more headers to added.
+                .WithExposedHeaders("X-Custom-Header") // Specify biz headers, x-custom-header as an example.
```
[the remark this answers is not in the document]
Comment (nets7snipper) [reply blotztask-api/Program.cs]: Removed.

File: blotztask-api/Program.cs (hunk 1 above)
Comment (nets7snipper) [reply blotztask-api/Program.cs]: done

Comment (sol-wizard) [reply blotztask-api/Program.cs]: nice but next time i will put it in the PR description

**Review by sol-wizard — APPROVED** [review]

**Review by sol-wizard — APPROVED** [review]

**Pull request merged**